app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java

```diff
@@ -161,7 +163,9 @@ else if (tempFavor.getStatusId() == ACCEPTED.toInt()) {
     } else { // not sure if this will be wrapped by completablefuture
       throw new IllegalStateException("Wrong Status");
     }
-    return updateFavorForCurrentUser(tempFavor, isRequested, -1);
+
+    return updateFavorForCurrentUser(tempFavor, isRequested, -1)
+      .thenCompose((aVoid -> UserUtil.getSingleInstance().settleTransaction(userIds.get(0), userIds.get(1), tempFavor.getReward())));
```

Comment:
Here I think the order of the first two arguments is not correct: the first one is the requester and the second one is the accepter (otherwise the accepter pays the requester!)

Reply:
I propose an alternative line of code, add another check too.
` .thenCompose(
                    (aVoid ->
                    {if(tempFavor.getStatusId() == FavorStatus.SUCCESSFULLY_COMPLETED.toInt())
                      return UserUtil.getSingleInstance().settleTransaction(userIds.get(1), userIds.get(0), tempFavor.getReward());
                     else return aVoid;
                    }));`